File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
#include <limits.h>

int maxProfit(int* arr, int n) {
    int min = INT_MAX;
    int maxProfit = 0;

    for (int i = 0; i < n; i++) {
        if(min>arr[i]){
            min = arr[i];
        }
        else if(arr[i] - min > maxProfit) {
            maxProfit = arr[i] - min;
        }
    }

    return maxProfit;
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit  = 0;
        int min = 21323423;
        for(int i : prices){
            if(i < min) min = i;
            else if(i - min > profit) profit = i - min;
        }
        return profit;
    }
}